src/Datadog.Trace/AppSec/Waf/NativeBindings/WafNative.cs

```diff
@@ -59,188 +58,189 @@ static Native()
+            DDWAF_LOG_LEVEL level,
             string function,
             string file,
             int line,
             string message,
             ulong message_len);
 
-        private delegate bool SetupLoggingDelegate(IntPtr cb, PW_LOG_LEVEL min_level);
+        private delegate bool SetupLoggingDelegate(IntPtr cb, DDWAF_LOG_LEVEL min_level);
 
-        private enum PW_LOG_LEVEL
+        private enum DDWAF_LOG_LEVEL
         {
-            PWL_TRACE,
-            PWL_DEBUG,
-            PWL_INFO,
-            PWL_WARN,
-            PWL_ERROR,
-            PWL_AFTER_LAST,
+            DDWAF_TRACE,
+            DDWAF_DEBUG,
+            DDWAF_INFO,
+            DDWAF_WARN,
+            DDWAF_ERROR,
+            DDWAF_AFTER_LAST,
         }
 
 #pragma warning disable SA1300 // Element should begin with upper-case letter
 
-        internal static PWVersion pw_getVersion()
-        {
-            return getVersion();
-        }
-
-        internal static IntPtr pw_initH(PWArgs wafRule, ref PWConfig config)
+        internal static IntPtr ObjectFreeFuncPtr
         {
-            return initH(wafRule, ref config);
+            get { return ObjectFreeFuncPtrField; }
         }
 
-        internal static void pw_clearRuleH(IntPtr wafHandle)
+        internal static DdwafVersionStruct GetVersion()
         {
-            clearRuleH(wafHandle);
+            return GetVersionField();
         }
 
-        internal static PWRet pw_runH(IntPtr wafHandle, PWArgs parameters, ulong timeLeftInUs)
+        internal static IntPtr Init(IntPtr wafRule, ref DdwafConfigStruct config)
         {
-            return runH(wafHandle, parameters, timeLeftInUs);
+            return InitField(wafRule, ref config);
         }
 
-        internal static void pw_freeReturn(PWRet output)
+        internal static void ResultFree(ref DdwafResultStruct output)
         {
-            freeReturn(output);
+            ResultFreeField(ref output);
         }
 
-        internal static IntPtr pw_initAdditiveH(IntPtr powerwafHandle)
+        internal static IntPtr InitContext(IntPtr powerwafHandle, IntPtr objFree)
         {
-            return initAdditiveH(powerwafHandle);
+            return InitContextField(powerwafHandle, objFree);
         }
 
-        internal static PWRet pw_runAdditive(IntPtr context, PWArgs newArgs, ulong timeLeftInUs)
+        internal static DDWAF_RET_CODE Run(IntPtr context, IntPtr newArgs, ref DdwafResultStruct result, ulong timeLeftInUs)
         {
-            return runAdditive(context, newArgs, timeLeftInUs);
+            return RunField(context, newArgs, ref result, timeLeftInUs);
         }
 
-        internal static void pw_clearAdditive(IntPtr context)
+        internal static void Destroy(IntPtr handle)
         {
-            clearAdditive(context);
+            DestroyField(handle);
         }
 
-        internal static PWArgs pw_getInvalid()
+        internal static void ContextDistroy(IntPtr handle)
```

Comment:
nit: Small typo, should be `ContextDestroy`

Reply:
fixed in bef5d961